src/main/host/descriptor/socket/inet/mod.rs

```diff
@@ -0,0 +1,251 @@
+use std::sync::Arc;
+
+use atomic_refcell::AtomicRefCell;
+
+use crate::cshadow as c;
+use crate::host::descriptor::{FileMode, FileState, FileStatus, SyscallResult};
+use crate::host::memory_manager::MemoryManager;
+use crate::host::syscall_types::{PluginPtr, SysCallReg, SyscallError};
+use crate::utility::callback_queue::CallbackQueue;
+use crate::utility::sockaddr::SockaddrStorage;
+use crate::utility::HostTreePointer;
+
+use tcp::TcpSocket;
+
+pub mod tcp;
+
+#[derive(Clone)]
+pub enum InetSocket {
+    Tcp(Arc<AtomicRefCell<TcpSocket>>),
```

Comment:
I think we should name the version that wraps the C TCP stack inside of it as something other than `Tcp`, and reserve the name `Tcp` for the new Rust TCP stack. That way when we ultimately remove the C stack we'll have the Rust name already set.